pkg/services/datasources/errors.go

```diff
@@ -11,4 +15,6 @@ var (
 	ErrDataSourceFailedGenerateUniqueUid = errors.New("failed to generate unique datasource ID")
 	ErrDataSourceIdentifierNotSet        = errors.New("unique identifier and org id are needed to be able to get or delete a datasource")
 	ErrDatasourceIsReadOnly              = errors.New("data source is readonly, can only be updated from configuration")
+	ErrDataSourceNameInvalid             = errutil.NewBase(errutil.StatusValidationFailed, "datasource.nameInvalid", errutil.WithPublicMessage("invalid datasource name"))
```

Comment:
the public message is what's is printed for the user so it should have proper puntuation:

```suggestion
	ErrDataSourceNameInvalid             = errutil.NewBase(errutil.StatusValidationFailed, "datasource.nameInvalid", errutil.WithPublicMessage("Invalid datasource name."))
```

Reply:
Got it. Updated both the error message